packages/common-all/src/types/configs/base.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * DendronConfigEntry
+ * Holds the value, label, and description of individual configuration entries.
+ *
+ * For config entries that can be an arbitrary value, only specify the label and description.
+ * For config entries that have pre-defined choices, provide the value as well as label and description specific to that value.
+ */
+export type DendronConfigEntry<T> = {
```

Comment:
since value is optional we can make `T=any` to cut down on boilerplate

Reply:
TIL 😅  still so much to learn about ts.